### DISCUSSION TITLE
Autoaccept cookies on some Region

### DISCUSSION THREAD

**Discussion by JiriBandura** [orestbida/cookieconsent#161] [Q&A] [answered]
Hi,
is there any way how to autoaccept all cookie category for some region?
For example, on "ES" (Spain) i want to keep default settings (everthing must be set as accept before initializing script) but for "SE" (Sweden) I want to autoaccept all categories. 

Make this work on GTM is easy, because it has the region value and GTM script dont have a 
<script type="text/plain" data-cookiecategory="analytics" >, but the other scripts have just that header and will not be removed when you arrive at the page, so the scripts are not running.

**Comment by orestbida** [comment, marked as answer]
Hi @JiriBandura,

yes, you can use the `.accept()` method to achieve this! I'm assuming you have some way of knowing what the current user's country is, and that this value is saved inside a variable — which I will simply call `country`.

Logic: if `cc_cookie` is not set and `country=='Sweden'` => accept all
```javascript
const cc = initCookieConsent();

cc.run({ /*...*/ });

const country = 'Sweden'; //value retrieved from server/api ?

// If is from Sweden and cookieconsent has never been accepted before => accept all by default
if (country  === 'Sweden' && !cc.validCookie('cc_cookie')) {
    cc.accept('all');
};
```

Note that if the current user is from Sweden and he/she does not make any explicit choice himself the very first time that the modal appears, there will be no more modal promtps on subsequent page loads as this implementation is literally the definition of **_"silence means consent"_**. They will however be able to change their preferences using the settings modal!

**Reply by JiriBandura** [reply]
Hi @orestbida ,

Thank you very much for your help. It is very useful. Even so, I would like to ask if a modal can be invoked when the user does not click on anything. I know it's a cc.show() function, but ideally, I need to combine that.

IN short.
1) Autoaccept all categories according to region (done)
2) Promtps modal until user action on anything inside modal (without forse_consent)

**Reply by orestbida** [reply]
Currently this is hard to achieve as there is no way to distinguish between the "user accept event" and "programmatic accept event". 

You would have to overwrite all of the modal's buttons functions with custom functions using custom code.

**Reply by orestbida** [reply]
@JiriBandura ,

I think adding a `mode` option might solve this issue. 

`country === 'Sweden'` => `mode: 'opt-out'`
`country !== 'Sweden'` => `mode: 'opt-in'` (current mode which is gdpr compliant)

If cookie is not set (first time) and `mode=='opt-out'` => automatically run scripts — whose categories are enabled by default — without asking for permission!

For this to work, you'd need to determine the `mode` before running the plugin!

```javascript

const country = 'Sweden';
let mode = 'opt-in';

if(country === 'Sweden') mode = 'opt-out';

// run plugin with your configuration
cc.run({
    // ...
    mode: mode,
    languages: {
        'en': {
            consent_modal: { /* ... */},
            settings_modal: {
                // ...
                blocks: [
                    // ...
                    {
                        // ...
                        toggle: {
                            value: 'analytics',     // cookie category
                            enabled: true,          // all scripts configured under 'analytics' category will be executed without consent
                            readonly: false
                        }
                    }
                ]
            }
        }
    }
});
```
Note that `mode` will stay optional, so if not specified, the `opt-in` mode will be used by default.
